File: backend/app/schemas/hotspot.py
import uuid
from datetime import datetime
from pydantic import BaseModel

class HotspotResponse(BaseModel):
    id: uuid.UUID
    title: str
    latitude: float
    longitude: float
    severity: str
    severity_score: float | None = None
    radius_meters: float | None = None
    reports_count: int
    complaint_ids: str | None = None
    dominant_category: str | None = None
    trend: str | None = None
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/hotspot.ts
export interface HotspotResponse {
  id: string;
  title: string;
  description?: string;
  latitude: number;
  longitude: number;
  severity: "high" | "medium" | "low" | string;
  severity_score?: number | null;
  radius_meters?: number | null;
  reports_count?: number;
  dominant_category?: string | null;
  trend?: string | null;
  is_active: boolean;
}
